executor/analyze.go

```diff
@@ -1371,24 +1453,40 @@ workLoop:
 				break workLoop
 			}
 			var collector *statistics.SampleCollector
+			totalMemInc := int64(0)
 			if task.isColumn {
 				if e.colsInfo[task.slicePos].IsGenerated() && !e.colsInfo[task.slicePos].GeneratedStored {
 					hists[task.slicePos] = nil
 					topns[task.slicePos] = nil
 					continue
 				}
-				sampleItems := make([]*statistics.SampleItem, 0, task.rootRowCollector.Base().Samples.Len())
+				sampleNum := task.rootRowCollector.Base().Samples.Len()
+				sampleItems := make([]*statistics.SampleItem, 0, sampleNum)
+				initMemSize := int64(unsafe.Sizeof(sampleItems)) + 8*int64(sampleNum) + statistics.EmptySampleItemSize*int64(sampleNum)
+				e.memTracker.Consume(initMemSize)
```

Comment:
Do we need to record `subBuildWorker` memusage? We didn't find it in top heap usage according to our test https://docs.google.com/document/d/1wBcSc-KeHK0o-jE6ykvf84dEj1-B1Esh1mYK7c3R_lQ/edit

Reply:
![image](https://user-images.githubusercontent.com/5151781/167100017-e9bffcd7-d7ee-4f98-868c-e3739533b591.png)